### DISCUSSION TITLE
python custom BSDF surface shader?

### DISCUSSION THREAD

**Discussion by foshyjoshy** [mitsuba-renderer/mitsuba3#1374] [Q&A]
Hello, im looking to create a custom python BSDF shader that displays a texture without any shading, lighting, or reflections.  I have looked into the docs, i don't see anything that already does this (Please tell me if I'm incorrect about this).  Would a BSDF like this work in Mitsuba with correct gradients?  

I have made this BSDF and it does give me the correct result when i render my scene, but if i run the 'Object pose estimation' tutorial with this custom shader, the angle and trans are not updated. If i remove my custom shader from the scene, everything works as should.

I know don't know much about BSDF's, so don't get scared. I just played around until something showed up in the render correctly. 

```py
class SurfaceShader(mi.BSDF):
    def __init__(self, props):
        mi.BSDF.__init__(self, props)
        self.texture = props['texture']
        self.m_components  = [mi.BSDFFlags.DiffuseReflection  |mi.BSDFFlags.FrontSide | mi.BSDFFlags.BackSide]
        self.m_flags = self.m_components[0]

    def sample(self, ctx, si, sample1, sample2, active):

        #### I have no idea what im doing here
        bs = mi.BSDFSample3f()
        bs.pdf = 1.0
        bs.sampled_component = 0
        bs.sampled_type = mi.BSDFFlags.DiffuseReflection
        bs.wo = si.wi
        bs.eta = 1.0
        texture_colour = self.texture.eval(si, active)

        return bs, texture_colour


    def eval(self, ctx, si, wo, active):
        return 0.0

    def pdf(self, ctx, si, wo, active):
        return 0.0

    def eval_pdf(self, ctx, si, wo, active):
        return  0.0, 0.0
    
    def traverse(self, callback):
        callback.put_parameter('texture', self.texture, mi.ParamFlags.NonDifferentiable)



```

**Comment by merlinND** [comment]
> shader that displays a texture without any shading, lighting, or reflections

A BSDF will by definition be affected by how much light is received by the surface, so it will "display" shadows and lighting.
Perhaps a setup that would better achieve this goal would be:
- `diffuse` BSDF with `reflectance` color `(0, 0, 0)` so that you have no reflections
- `area` emitter inside of the shape, with `radiance` being the texture you want to display.

I don't know how it would play out with the "object pose estimation" tutorial, but it's worth a try.
